[2022-11-19T00:00:45.409+0000] {taskinstance.py:1165} INFO - Dependencies all met for <TaskInstance: spark_operator.spark-job scheduled__2022-11-18T00:00:00+00:00 [queued]>
[2022-11-19T00:00:45.443+0000] {taskinstance.py:1165} INFO - Dependencies all met for <TaskInstance: spark_operator.spark-job scheduled__2022-11-18T00:00:00+00:00 [queued]>
[2022-11-19T00:00:45.444+0000] {taskinstance.py:1362} INFO - 
--------------------------------------------------------------------------------
[2022-11-19T00:00:45.444+0000] {taskinstance.py:1363} INFO - Starting attempt 1 of 1
[2022-11-19T00:00:45.445+0000] {taskinstance.py:1364} INFO - 
--------------------------------------------------------------------------------
[2022-11-19T00:00:45.472+0000] {taskinstance.py:1383} INFO - Executing <Task(SparkSubmitOperator): spark-job> on 2022-11-18 00:00:00+00:00
[2022-11-19T00:00:45.486+0000] {standard_task_runner.py:82} INFO - Running: ['***', 'tasks', 'run', 'spark_operator', 'spark-job', 'scheduled__2022-11-18T00:00:00+00:00', '--job-id', '57', '--raw', '--subdir', 'DAGS_FOLDER/load_data_to_spark.py', '--cfg-path', '/tmp/tmp_tmqii0h']
[2022-11-19T00:00:45.487+0000] {standard_task_runner.py:83} INFO - Job 57: Subtask spark-job
[2022-11-19T00:00:45.482+0000] {standard_task_runner.py:55} INFO - Started process 16109 to run task
[2022-11-19T00:00:45.606+0000] {task_command.py:376} INFO - Running <TaskInstance: spark_operator.spark-job scheduled__2022-11-18T00:00:00+00:00 [running]> on host 71a129efd3d8
[2022-11-19T00:00:45.745+0000] {taskinstance.py:1592} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_OWNER=***
AIRFLOW_CTX_DAG_ID=spark_operator
AIRFLOW_CTX_TASK_ID=spark-job
AIRFLOW_CTX_EXECUTION_DATE=2022-11-18T00:00:00+00:00
AIRFLOW_CTX_TRY_NUMBER=1
AIRFLOW_CTX_DAG_RUN_ID=scheduled__2022-11-18T00:00:00+00:00
[2022-11-19T00:00:45.763+0000] {base.py:71} INFO - Using connection ID 'spark_local' for task execution.
[2022-11-19T00:00:45.765+0000] {spark_submit.py:334} INFO - Spark-Submit cmd: spark-submit --master spark://Posathorns-MacBook-Pro-4.local:7077 --name arrow-spark json_to_spark.py
[2022-11-19T00:00:45.810+0000] {spark_submit.py:485} INFO - /home/***/.local/bin/spark-submit: line 27: /opt/homebrew/Cellar/apache-spark/3.3.1/libexec/bin/spark-class: No such file or directory
[2022-11-19T00:00:45.828+0000] {taskinstance.py:1851} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 157, in execute
    self._hook.submit(self._application)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 417, in submit
    f"Cannot execute: {self._mask_cmd(spark_submit_cmd)}. Error code is: {returncode}."
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://Posathorns-MacBook-Pro-4.local:7077 --name arrow-spark json_to_spark.py. Error code is: 127.
[2022-11-19T00:00:45.835+0000] {taskinstance.py:1406} INFO - Marking task as FAILED. dag_id=spark_operator, task_id=spark-job, execution_date=20221118T000000, start_date=20221119T000045, end_date=20221119T000045
[2022-11-19T00:00:45.846+0000] {standard_task_runner.py:105} ERROR - Failed to execute job 57 for task spark-job (Cannot execute: spark-submit --master spark://Posathorns-MacBook-Pro-4.local:7077 --name arrow-spark json_to_spark.py. Error code is: 127.; 16109)
[2022-11-19T00:00:45.882+0000] {local_task_job.py:159} INFO - Task exited with return code 1
[2022-11-19T00:00:45.899+0000] {taskinstance.py:2623} INFO - 0 downstream tasks scheduled from follow-on schedule check
